src/pages/RequestCallPage.js

```diff
@@ -104,26 +100,13 @@ const defaultProps = {
 class RequestCallPage extends Component {
     constructor(props) {
         super(props);
-        const {firstName, lastName} = PersonalDetails.extractFirstAndLastNameFromAvailableDetails(props.currentUserPersonalDetails);
-        this.state = {
-            firstName,
-            hasFirstNameError: false,
-            lastName,
-            phoneNumber: this.getPhoneNumber(props.loginList) || '',
-            phoneExtension: '',
-            phoneExtensionError: '',
-            hasLastNameError: false,
-            phoneNumberError: '',
-            onTheWeekend: false,
-        };
+        this.name = PersonalDetails.extractFirstAndLastNameFromAvailableDetails(props.currentUserPersonalDetails);
+        this.isWeekend = moment().day() === 0 || moment().day() === 6;
+        this.isBlockedFromConcierge = User.isBlockedFromConcierge(props.blockedFromConcierge);
```

Comment:
It feels like we are making the app slightly less reactive with the approach of setting things in the constructor that are computed from props that could later change. Kind of edge case, but e.g. if the user is blocked in realtime, updates the personal details, or the day changes while the component is mounted then we will have stale values here? Maybe fine to just grab them from props when we are ready to use them.

Reply:
Yea, I considered that and deemed it to be a rare edge case. Having said that, making it more reactive is pretty easy so it makes sense to do it!